[11.0] mig partner contact in several companies

njeudy opened the pull request:
- Migrate partner_contact_in_several_companies 
- Migrate partner_contact_personal_information_page

In the same PR because dependency

Commit:
Create partner_contact_personal_information_page.

Really I'm just renaming partner_contact_base.
The new name is more self-explanatory.

Commit:
Deprecate base_contact.

- Remove nationality and birthday features.
- Move the remaining features to partner_contact_in_several_companies.
- Make base_contact a dummy module.
- Warn deprecation everywhere possible.

sysadminmatmoz committed:
Translations and templates sync

yvaucher committed:
Add missing default oca icons

yvaucher committed:
Add missing default oca icons

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] prefix versions with 8.0

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

Commit:
partner_contact_personal_information updated to version 9.0

Commit:
OCA module upgrade to version 9.0.

- Remove Domain Error on contact view.
- Additionally, improve behaviour of personal contact info page.
- Change to README.
- No longer dependent on partner_contact_personal_information_page
- Added tests for ir_actions
- Include extra demo data.
- Increase test coverage.
- Remove birthdate from demo data.

yvaucher committed:
Making the 'personal contact' informations depentens of the module 'personal_contact_information'

leemannd committed:
Remove refecence of base in xmlid

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

pedrobaeza committed:
[MIG] Rename manifest files

Commit:
[MIG] Migration partner_contact_birthdate partner_contact_gender part… (#315)

[MIG] Migration partner_contact_birthdate partner_contact_gender partner_contact_personal_information_page to v10

Commit:
[MIG] 10.0 partner_contact_in_several_companies (#347)

* [MIG] inital work to 10.0

* [FIX] restore computed field

- select=1 is not usable anymore ...

* [FIX] convert _fields_sync to new api

* [FIX] restore form . import statement

* [FIX] readme with last template.

* [FIX] import statement.

* [FIX] check all tests

* [FIX] Flake8 errors fixes

* [FIX] flake 8

* [FIX] wrong obj call self -> self.contact_id

* [FIX] typo

* [FIX] flake 8

* [FIX] add index on contact_type

* [fix] use in statement on test

* [FIX] update base action with new context

* [FIX] test fixes

* [FIX] test work

* [FIX] flake 8 over indent

* [FIX] fixes on lastest comments

- better readability
- remove <data></data> tags

* [FIX] model that same pattern for _fields_sync

- add test ensure_once()

* [FIX] remove last data xml tags

* [FIX] fix action test

* [FIX] typo in import statement

* [FIX] remove logger from tests

* [FIX] new flake8 compute method name for OCA

* [FIX] minor typo

* [FIX] partner_contact_in_several_companies: remove contributor key in manifest

* [FIX] partner_contact_in_several_companies: residual onchange on partner view

* Revert "[FIX] partner_contact_in_several_companies: remove contributor key in manifest"

This reverts commit 56e511a9a7e1dd20cf99578a340a00301a98ba85.

* [FIX] partner_contact_personal_information_page: clean manifest and authors

* [FIX] partner_contact_in_several_companies: required = True not needed

njeudy commented:
I think a need to squash :)

But don't know how ?

njeudy commented:
@pedrobaeza @gurneyalex don't understand runbot and codecov red bullet ... some help ?

etobella commented:
The problem with the runbot is that your code raises a WARNING:
`2017-10-24 20:44:42,319 151 WARNING openerp_test odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view res.partner.kanban.contact (): use the hasclass(*classes) function to filter elements by their classes`
Codecov error will be solved when some PR are accepted (no initial report can be found). It should not be a major problem.

njeudy commented:
Ok will change this :) thanks @etobella

njeudy commented:
- Ok xpath expression changed
- And squash transbot commit

njeudy commented:
@pedrobaeza all green :) ready for review ?

pedrobaeza commented:
Well, I'm not directly interested in this module, so let's wait for other interested reviewers. Any way, I think if you rebase over current branch, you will get coverage, which is a good indicator also.

njeudy commented:
@pedrobaeza ok no probleme will rebase. thanks

njeudy commented:
@Yajo @pedrobaeza what should I do to pass codecov ? it is just 10.0 to 11.0 migration no improvement or new function ...

yajo commented:
In this case codecov indicates that some merged code is not being tested. If you hit "Details > Diff", you'll see some red lines indicating that. You'd have to add some test for those lines to go over the target patch coverage (96.34%) and to avoid lowering the current project coverage ratio.

It would be a much appreciated effort, although I wouldn't require such thing IMHO.

njeudy commented:
@Yajo I add some test but coverage checks was not launched .. ?

njeudy commented:
@Yajo done ;)

yajo reviewed (approved):
Cool! Please squash all migration-related commits into one for the final merge 😊

njeudy commented:
@Yajo like that ?

yajo commented:
With https://github.com/OCA/partner-contact/pull/507/commits/867d552bdeb4712d34ba218f2c992650aaa6ce5b too

njeudy commented:
@Yajo : did not understand why this commit is shown, in my branch all commit are squashed in one.
https://github.com/njeudy/partner-contact/commits/11.0-mig-partner_contact_in_several_companies ... don't know how to do.

yajo commented:
@njeudy don't worry, let me show you. Look at the commits in the PR:

![captura el 2017-11-15 a las 14 36 21](https://user-images.githubusercontent.com/973709/32838846-9da18e8a-ca12-11e7-915f-22a39d01193c.png)

I highlighted all of the remaining translation commits.

The normal flow would be:

```
git checkout 11.0-mig-partner_contact_in_several_companies
git rebase --interactive 11.0
```

That should open the editor you have configured and show all the list of commits from your branch.

They should appear in this fashion:

```
pick <commit-sha> <commit-msg>
[...]
```

Now you should cut all those lines that I marked in red and move them to the bottom, just before your migration commit, and mark as `fixup` all of them except the first one. The end of the commits list should look something similar to this:

```
pick e112903 Create partner_contact_personal_information_page.
[...]
# Pick 1st translation commit
pick 4d5d348 OCA Transbot updated translations from Transifex 
# Fixup the rest of translation commits
fixup e639882 OCA Transbot updated translations from Transifex
# ...
fixup 2024a3e OCA Transbot updated translations from Transifex
# Your migration commit at the end:
pick 43e65d6 [MIG] V11 migration
```

Then save and exit. Git will start picking all code commits, then it will pick 1st translation commit and fixup (squash discarding commit message) all of other translation commits in that one, then it will pick your migration commit and will let a clean branch! There's chance that merge conflicts appear, in such case install `meld` and, when git rebase stops, do:

```
git mergetool
# now, using meld, fix conflicts on middle pane. Then:
git rebase --continue
```

If you even use that fixup system to remove the duplicate commits from your branch, that would be even more awesome, but this should be just fine with that.

njeudy commented:
@Yajo thanks for your help, now it should be ok and have learn a lot if git rebase tricks :)

yajo commented:
Yikes, I know you're gonna kill me, but just this little thing remaining in the rebase:

![captura de pantalla de 2017-11-20 09-42-10](https://user-images.githubusercontent.com/973709/33009431-2193a3b4-cdd7-11e7-9783-53b33ecba0c2.png)

Remove the 1st of those 2 commits please. It's something that should have never happened. In fact, if you check it out you'll see it's empty: https://github.com/OCA/partner-contact/pull/507/commits/88e8d3c83c1707db573ecece0dbe6e709023b18a

To remove a commit in a `git rebase -i`, just delete or comment the commit line.

njeudy commented:
@Yajo ok done :) thanks for the help and clean review.

njeudy commented:
@Yajo @pedrobaeza ready to merge now ?

simahawk commented on partner_contact_in_several_companies/__init__.py:
```diff
@@ -0,0 +1,4 @@
+# -*- coding: utf-8 -*-
```
aren't we dropping utf8 headers? Let's take a decision :smile: I have some pending PRs where I've already removed them and some to be updated... Just want to make sure we are doing the same thing for all the PRs/modules

njeudy replied:
on all .py files ?

simahawk commented on partner_contact_in_several_companies/__manifest__.py:
```diff
@@ -0,0 +1,26 @@
+    "summary": "Allow to have one contact in several partners",
+    "version": "11.0.1.0.0",
+    "category": "Customer Relationship Management",
+    "website": "https://odoo-community.org/",
```
website should be https://github.com/OCA/partner-contact

njeudy replied:
done

simahawk commented on partner_contact_in_several_companies/__manifest__.py:
```diff
@@ -0,0 +1,26 @@
+    "website": "https://odoo-community.org/",
+    "author": "Nicolas JEUDY,Odoo Community Association (OCA),Odoo SA",
+    "license": "AGPL-3",
+    'application': False,
```
you can remove application, installable and auto_install

njeudy replied:
done

simahawk commented on partner_contact_personal_information_page/README.rst:
```diff
@@ -0,0 +1,86 @@
+
+* None.
+
+Bug Tracker
```
update this part w/ new layout https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

njeudy replied:
Done

simahawk commented on partner_contact_personal_information_page/i18n/partner_contact_personal_information_page.pot:
```diff
@@ -0,0 +1,22 @@
+# Translation of Odoo Server.
```
aren't we excluding POT files?

pedrobaeza replied:
POT files indeed should be removed

njeudy replied:
done

simahawk commented:
A consideration about history: can't we squash little commits from the same authors? For instance:
![image](https://user-images.githubusercontent.com/347149/33478100-6fc6a9a0-d688-11e7-8b8c-728ba0dc9a82.png)
Despite the fact that these commits should have be squashed in their PR we still have the chance to clean them up. 
We can group them w/ generic commits like "$name_of_the_module: misc fix/imp" and keep the sub-commits in the extended commit msg (as I did in #526).

Commit:
[MIG] V11 migration

- fax field removed from res.partner
- 2to3 on all py files
- add contributor and update README
- use hasclass in place of @class in xpath expr
- improve tests

[FIX] typo and clean README

njeudy commented:
@simahawk : should I squash commit from same author ?

simahawk commented:
@njeudy as long as they are not really important like "Introduced breaking change in API"... I'd say yes :)
But all the small changes I see there can be squashed by author for sure.

njeudy commented:
@simahawk Ok done squash commit from same user in one

simahawk reviewed (approved):
LGTM. I'd squash commits like "make module uninstallable" and alike, just because they are useless in this context. But is not a blocker and we don't have guidelines for this ;)

njeudy commented:
@pedrobaeza @simahawk @Yajo Ok to merge ?

pedrobaeza commented:
This has broken Travis. I have already fixed one issue, but it seems that there are more. This can't be installed. Please check https://travis-ci.org/OCA/partner-contact/jobs/317716741#L780

pedrobaeza commented:
Please fix Travis after merging this.

pedrobaeza commented:
Solved thanks to @drdran
